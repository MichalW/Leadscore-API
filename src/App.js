import React from 'react';
import {Container} from 'semantic-ui-react';
import {Provider} from 'react-redux';

import configureStore from './store';
import './App.css';

const store = configureStore();

const App = () => (
  <Provider store={store}>
    <div className="App">
      <Container>
        Welcome to LeadScore
      </Container>
    </div>
  </Provider>
);

export default App;
