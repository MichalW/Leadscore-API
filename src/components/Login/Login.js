import React from 'react';
import {
  Button, Container, Form, Grid, Input, Label,
} from 'semantic-ui-react';

const Login = () => (
  <Container>
    <Grid columns={3} centered>
      <Grid.Row>
        <Grid.Column>
          <Form>
            <Form.Field>
              <Label>Login</Label>
              <Input placeholder="Login" />
            </Form.Field>

            <Form.Field>
              <Label>Password</Label>
              <input placeholder="Password" type="password" />
            </Form.Field>

            <Button type="submit">Login</Button>
          </Form>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Container>
);

export default Login;
