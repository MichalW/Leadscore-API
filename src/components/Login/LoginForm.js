import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Label, Segment,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = (event) => {
    this.setState({username: event.target.value});
  };

  onChangePassword = (event) => {
    this.setState({password: event.target.value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field>
            <Label>Login</Label>
            <Input placeholder="Login" onChange={this.onChangeUsername} value={username} />
          </Form.Field>

          <Form.Field>
            <Label>Password</Label>
            <input placeholder="Password" type="password" onChange={this.onChangePassword} value={password} />
          </Form.Field>

          <Button type="submit" fluid primary>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
