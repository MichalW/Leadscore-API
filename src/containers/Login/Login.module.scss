@import "../../styles/colors";

.wrapper {
  background-color: $color-blue;
  min-height: 100vh;
}
