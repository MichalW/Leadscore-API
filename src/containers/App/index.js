import {connect} from 'react-redux';

import App from './App';

const mapStateToProps = (state) => ({
  signedIn: state.authentication.signedIn,
});

export default connect(mapStateToProps)(App);
