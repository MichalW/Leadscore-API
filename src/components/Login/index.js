import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Login from './Login';
import {signInRequest} from '../../actions/authentication';

const mapDispatchToProps = dispatch => ({
  signInRequest: bindActionCreators(signInRequest, dispatch),
});

export default connect(null, mapDispatchToProps)(Login);
