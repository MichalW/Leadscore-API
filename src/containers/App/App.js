import PropTypes from 'prop-types';
import React from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';

import Login from '../../components/Login';
import Home from '../../components/Home';
import Contacts from '../../components/Contacts';
import Page404 from '../../components/Page404';

const App = ({signedIn}) => {
  if (!signedIn) {
    return (<Login />);
  }

  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/contacts" component={Contacts} />
        <Route component={Page404} />
      </Switch>
    </Router>
  );
};

App.propTypes = {
  signedIn: PropTypes.bool.isRequired,
};

export default App;
