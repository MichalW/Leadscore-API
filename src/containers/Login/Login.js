import React from 'react';
import {Grid, Header} from 'semantic-ui-react';

import LoginForm from '../../components/LoginForm';
import styles from './Login.module.scss';

const Login = () => (
  <Grid
    centered
    className={styles.wrapper}
    columns={3}
    style={{minHeight: '100vh'}}
    verticalAlign="middle"
  >
    <Grid.Row>
      <Grid.Column>
        <Header as="h1">
          <strong>Lead</strong>
          score
        </Header>

        <LoginForm />
      </Grid.Column>
    </Grid.Row>
  </Grid>
);

export default Login;
