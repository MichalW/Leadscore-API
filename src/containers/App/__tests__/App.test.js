import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import App from '../App';
import Login from '../../Login';

it('should render App signed out', () => {
  const props = {
    signedIn: false,
    username: 'ExampleUsername',
  };

  const layout = shallow(<App {...props} />);

  expect(layout.find(Login).exists()).toBe(true);
  expect(toJson(layout)).toMatchSnapshot();
});

it('should render App signed in', () => {
  const props = {
    signedIn: true,
    username: 'ExampleUsername',
  };

  const layout = shallow(<App {...props} />);

  expect(toJson(layout)).toMatchSnapshot();
});
