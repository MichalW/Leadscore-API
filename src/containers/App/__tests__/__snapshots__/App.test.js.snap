// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App signed in 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/contacts"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;

exports[`should render App signed out 1`] = `<Connect(Login) />`;
